sonar/modules/documents/utils.py

```diff
@@ -63,3 +70,164 @@ def publication_statement_text(provision_activity):
         value = remove_trailing_punctuation(value)
         statement_with_language[key] = value
     return statement_with_language
+
+
+def get_current_organisation_code():
+    """Return current organisation by globals or query parameter."""
+    # Organisation is present in query parameters, useful for API calls.
+    organisation = request.args.get('view')
+    if organisation:
+        return organisation
+
+    # Organisation stored in globals
+    if g.get('organisation', {}).get('code'):
+        return g.organisation['code']
+
+    # Default organisation
+    return current_app.config.get('SONAR_APP_DEFAULT_ORGANISATION')
+
+
+def get_file_links(file, record):
+    """Return link data to preview and/or download the file.
+
+    :param file: File record.
+    :param record: Record.
+    :returns: Dict containing the URL, the download URL and the type of link.
+    """
+    links = {'external': None, 'preview': None, 'download': None}
+
+    # File is restricted, no link.
+    if file['restriction']['restricted']:
+        return links
+
+    # File has an external url
+    if record['external_url'] and file.get('external_url'):
+        links['external'] = file['external_url']
+        return links
+
+    match = re.search(r'\.(.*)$', file['key'])
+    if not match:
+        return links
+
+    links['download'] = '/documents/{pid}/files/{key}'.format(
```

Comment:
This can be replaced probably with an `url_for`

Reply:
No, because it can be called by the REST application and the route does not exists..